shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -778,4 +767,18 @@ void FlutterWindowsEngine::OnDwmCompositionChanged() {
   view_->OnDwmCompositionChanged();
 }
 
+void FlutterWindowsEngine::OnServiceBindingsRegistered() {
```

Comment:
I'm a bit concerned that in the future we may have subtle bugs due to unexpected ordering of top-level message handlers. What do you think of always registering the lifecycle manager but updating the lifecycle manager to ignore `WM_CLOSE` messages until it is notified that the service bindings were registered?

Reply:
Another concern that would be addressed: we should only register the handler once even if the framework sends multiple `System.registerBinding` messages.